src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,255 @@
+                calculator.calculate(first, second, '+'),
+                DELTA);
+    }
+
+    @Test
+    void addingTwoNegative_Ok() {
+        assertEquals(-first + -second,
+                calculator.calculate(-first, -second, '+'),
+                DELTA);
+    }
+
+    @Test
+    void addingNegativeAndPositive_Ok() {
+        assertEquals(-first + second,
+                calculator.calculate(-first, second, '+'),
+                DELTA);
+    }
+
+    @Test
+    void addingWithZeroAtFirstPlace_Ok() {
+        assertEquals(0 + second,
+                calculator.calculate(0, second, '+'),
+                DELTA);
+    }
+
+    @Test
+    void addingWithZeroAtSecondPlace_Ok() {
+        assertEquals(first + 0,
+                calculator.calculate(first, 0, '+'),
+                DELTA);
+    }
+
+    @Test
+    void addingToDoubleMaxValue_Ok() {
+        assertEquals(Double.MAX_VALUE + second,
+                calculator.calculate(Double.MAX_VALUE, second, '+'),
+                DELTA);
+    }
+
+    @Test
+    void dividingTwoPositive_Ok() {
+        assertEquals(first / second,
+                calculator.calculate(first, second, '/'),
+                DELTA);
+    }
+
+    @Test
+    void dividingTwoNegative_Ok() {
+        assertEquals(-first / -second,
+                calculator.calculate(-first, -second, '/'),
+                DELTA);
+    }
+
+    @Test
+    void dividingPositiveAndNegative_Ok() {
+        assertEquals(first / -second,
+                calculator.calculate(first, -second, '/'),
+                DELTA);
+    }
+
+    @Test
+    void dividingWithZeroAtFirstPlace_Ok() {
+        assertEquals(0 / -second,
+                calculator.calculate(0, -second, '/'),
+                DELTA);
+    }
+
+    @Test
+    void dividingWithZeroAtSecondPlace_NotOk() {
+        assertThrows(ArithmeticException.class, () -> calculator.calculate(first, 0, '/'));
+    }
+
+    @Test
+    void dividingDoubleMaxValue_Ok() {
+        assertEquals(Double.MAX_VALUE / second,
+                calculator.calculate(Double.MAX_VALUE, second, '/'),
+                DELTA);
+    }
+
+    @Test
+    void dividingByDoubleMaxValue_Ok() {
+        assertEquals(first / Double.MAX_VALUE,
+                calculator.calculate(first, Double.MAX_VALUE, '/'),
+                DELTA);
+    }
+
+    @Test
+    void multiplicationTwoPositive_Ok() {
+        assertEquals(first * second,
+                calculator.calculate(first, second, '*'),
+                DELTA);
+    }
+
+    @Test
+    void multiplicationTwoNegative_Ok() {
+        assertEquals(-first * -second,
+                calculator.calculate(-first, -second, '*'),
+                DELTA);
+    }
+
+    @Test
+    void multiplicationPositiveAndNegative_Ok() {
+        assertEquals(first * -second,
+                calculator.calculate(first, -second, '*'),
+                DELTA);
+    }
+
+    @Test
+    void multiplicationWithZeroAtFirstPlace_Ok() {
+        assertEquals(0 * second,
```

Comment:
в тестах краще відразу вказувати кінцевий результат обчислень який ми плануємо отримати. 
Ми можемо лишити обчислення в тих тестах, де важко порахувати вручну результат. Але наприклад ми точно знаємо що при множені число на нуль ми завжди отримаємо нуль
```suggestion
        assertEquals(0,
```